dotcom-rendering/src/footballTables.ts

```diff
@@ -1,36 +1,150 @@
-export type TeamResultScore = {
-	name: string;
-	score: number;
-};
+import { listParse, type ParserError, type TeamScore } from './footballMatches';
+import type {
+	FEFootballTable,
+	FEGroup,
+	FELeagueTableEntry,
+	FETeamResult,
+} from './frontend/feFootballTablesPage';
+import type { Result } from './lib/result';
+import { ok } from './lib/result';
 
 export type TeamResult = {
+	matchId: string;
+	self: TeamScore;
+	foe: TeamScore;
+};
+
+type TeamInfo = {
+	name: string;
 	id: string;
-	self: TeamResultScore;
-	foe: TeamResultScore;
+	url?: string;
+};
+
+type Entry = {
+	position: number;
+	team: TeamInfo;
+	gamesPlayed: number;
+	won: number;
+	drawn: number;
+	lost: number;
+	goalsFor: number;
+	goalsAgainst: number;
+	goalDifference: number;
+	points: number;
+	results: TeamResult[];
 };
 
 export type FootballTableCompetition = {
 	url: string;
 	name: string;
 	hasGroups: boolean;
+	dividers: number[];
 	tables: FootballTableData[];
 };
 
 export type FootballTableData = {
-	groupName: string;
-	dividers: number[];
-	entries: {
-		position: number;
-		team: { name: string; id: string; url: string };
-		gamesPlayed: number;
-		won: number;
-		drawn: number;
-		lost: number;
-		goalsFor: number;
-		goalsAgainst: number;
-		goalDifference: number;
-		points: number;
-		results: TeamResult[];
-	}[];
-	linkToFullTable: boolean;
+	groupName?: string;
+	entries: Entry[];
+};
+
+export type FootballTableCompetitions = FootballTableCompetition[];
+
+const parseTables = (
+	feGroups: FEFootballTable['groups'],
+): Result<ParserError, FootballTableData[]> => {
+	const groupsParser = listParse(parseTable);
+	return groupsParser(feGroups);
+};
+
+const parseTable = (
+	feGroup: FEGroup,
+): Result<ParserError, FootballTableData> => {
+	const entriesParser = listParse(parseEntry);
+	const parsedEntries = entriesParser(feGroup.entries);
+
+	if (parsedEntries.kind === 'error') {
+		return parsedEntries;
+	}
+
+	return ok({
+		groupName: feGroup.round.name,
+		entries: parsedEntries.value.sort((a, b) => a.position - b.position),
+	});
+};
+
+const parseResult = (result: FETeamResult): Result<ParserError, TeamResult> => {
+	return ok({
+		matchId: result.matchId,
+		self: {
+			id: result.self.id,
+			name: result.self.name,
+			score: result.self.score,
+		},
+		foe: {
+			id: result.foe.id,
+			name: result.foe.name,
+			score: result.foe.score,
+		},
+	});
+};
+
+const parseResults = (
+	teamResults: FETeamResult[],
+): Result<ParserError, TeamResult[]> => {
+	const resultsParser = listParse(parseResult);
+	return resultsParser(teamResults);
 };
```

Comment:
Similar to above, this could be more simply written as:

```suggestion
const parseResults = listParse(parseResult);
```

Reply:
Done